proposed/http-client/http-client.md

```diff
@@ -46,6 +46,10 @@ interface HttpClient
      *
      * If a request is sent without any prior configuration, a HTTP client MUST NOT follow redirects.
      *
+     * The client MAY do motifications to the Request before sending it. Because of PSR-7 objects are
+     * immutable one cannot assume that the object passed to HttpClient::sendRequest will have the same
+     * reference as the Request object returned by an exception.
```

Comment:
I would not mention exception specifically as the user might access the messages  from other sources. I would rather say something more general and mention exceptions as an example.